rfc/ecs_e2e_tests.md

```diff
@@ -0,0 +1,372 @@
+the Spinnaker AWS provider. 
+
+However, ECS server groups do typically _utilize_ both - load balancer 
+health checks being most common (vs. provider), and security groups 
+being required for launching on AWS Fargate. Therefore testing ECS 
+server groups will require prefiguring at least one (security group) 
+or both resources outside of ECS-specific code paths. At runtime 
+this could be done by invoking the AWS provider (not recommended, 
+duplicates behavior under test) or directly via the `aws-cli` 
+(already available in the test env).
+
+Alternatively, these resources can be created once in the hosting account 
+and reused between tests (preferred; discussed further down).
+
+
+#### Test Cases
+
+Given the above, the test case most representative of ongoing ECS provider 
+use would:
+
+1. Use an artifact to define a container application
+2. Use an application load balancer + healthchecks
+
+Subsequent major test cases to cover in an end-to-end way include using 
+the older `targetGroup` fields vs. the newer `targetGroupMappings`, 
+using provider health checks intead of load balancer, and with service 
+discovery backed by [AWS Cloud Map](https://aws.amazon.com/cloud-map/). 
+These represent known customer use cases and cover forking code 
+paths in the ECS provider deployment logic.
+
+##### Prioritized list of initial test cases:
+
+1. **Create ECS server group with an artifact, using** 
+   **load balancer health checks.**
+    * First e2e test case to add; design discussed below.
+2. Create ECS server group with direct inputs, using 
+   load balancer health checks and "legacy" target group fields.
+3. Create ECS server group with an artifact that defines container health 
+   checks, using provider health checks and service discovery.
+
+
+### Required changes
+
+#### Actual ECS e2e test workflow
+
+The first test would be modeled after the 
+[google_server_group_test](https://github.com/spinnaker/buildtool/blob/master/testing/citest/tests/google_server_group_test.py) 
+in structure and the [aws_smoke_test](https://github.com/spinnaker/buildtool/blob/master/testing/citest/tests/aws_smoke_test.py) 
+w/r/t observers.
+
+High-level test workflow:
+
+0. ASSUMES: load balancer, target group, and S3 bucket available in account or otherwise bootstrapped.
+1. Create a Spinnaker application with the AWS and ECS providers enabled.
+2. Create a server group (step also uploads JSON file to S3).
+3. Resize the server group.
+4. Destroy the server group (and the uploaded artifact).
+5. Delete the application.
+
+* Test psuedo code w/ annotations re: implementation questions is 
+available [here](https://github.com/allisaurus/buildtool/blob/ecs-test/testing/citest/tests/ecs_server_group_test.py), 
+for context.
+
+
+#### Test environment : AWS Account changes
+
+The ECS provider itself does not own any credentials, but points to a configured 
+AWS provider account to deploy to. The ECS provider can leverage an 
```

Comment:
Use of existing AWS test account based on conversations w/ @plumpy ; Ideally it would be managed by the CDF prior to implementation.